airbyte-integrations/connectors/source-marketo/source_marketo/source.py

```diff
@@ -0,0 +1,530 @@
+    Return string with dict, example: "Completed"
+    """
+
+    def path(self, **kwargs) -> str:
+        return f"/bulk/v1/{self.stream_name}/export/{self.export_id}/status.json"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> List[str]:
+        return [response.json()[self.data_field][0]["status"]]
+
+
+class Leads(MarketoExportBase):
+    """
+    Return list of all leeds.
+    API Docs: http://developers.marketo.com/rest-api/bulk-extract/bulk-lead-extract/
+    """
+
+    cursor_field = "updatedAt"
+
+    def __init__(self, config: Mapping[str, Any]):
+        super().__init__(config, "leads")
+
+    @property
+    def stream_fields(self):
+        return list(self.get_json_schema()["properties"].keys())
+
+
+class Activities(MarketoExportBase):
+    """
+    Base class for all the activities streams,
+    provides functionality for dynamically created classes as streams of data.
+    API Docs: http://developers.marketo.com/rest-api/bulk-extract/bulk-activity-extract/
+    """
+
+    primary_key = "marketoGUID"
+    cursor_field = "activityDate"
+
+    def __init__(self, config: Mapping[str, Any]):
+        super().__init__(config, "activities")
+
+    @property
+    def stream_filter(self):
+        return {"activityTypeIds": [self.activity["id"]]}
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        properties = {
+            "marketoGUID": {"type": ["null", "string"]},
+            "leadId": {"type": ["null", "integer"]},
+            "activityDate": {"type": ["null", "string"], "format": "date-time"},
+            "activityTypeId": {"type": ["null", "integer"]},
+            "campaignId": {"type": ["null", "integer"]},
+            "attributes": {"type": ["null", "string"]},
+        }
+
+        if "primaryAttribute" in self.activity:
+            properties["primaryAttributeValue"] = {"type": ["null", "string"]}
+            properties["primaryAttributeName"] = {"type": ["null", "string"]}
+            properties["primaryAttributeValueId"] = {"type": ["null", "string"]}
+
+        if "attributes" in self.activity:
+            for attr in self.activity["attributes"]:
+                attr_name = clean_string(attr["name"])
+
+                if attr["dataType"] in ["datetime", "date"]:
+                    field_schema = {"type": "string", "format": "date-time"}
+                elif attr["dataType"] in ["integer", "percent", "score"]:
+                    field_schema = {"type": "integer"}
+                elif attr["dataType"] in ["float", "currency"]:
+                    field_schema = {"type": "number"}
+                elif attr["dataType"] == "boolean":
+                    field_schema = {"type": "boolean"}
+                elif attr["dataType"] in STRING_TYPES:
+                    field_schema = {"type": "string"}
+                elif attr["dataType"] in ["array"]:
+                    field_schema = {"type": "array", "items": {"type": ["integer", "number", "string", "null"]}}
```

Comment:
This could lead to validation_errors in the future, avoid to specify the complex types.